docs/topics/upgrade.md

```diff
@@ -0,0 +1,156 @@
+
+```
+export latest_kube="<latest patch version of kubernetes>"
+```
+
+* For `coredns` check the latest patch version.
+
+Find your current version:
+
+```
+kubectl -n kube-system get deploy coredns -o jsonpath='{.spec.template.spec.containers[0].image}' | cut -d':' -f2-
+```
+
+You can see the latest patch version on the CoreDNS release page: https://github.com/coredns/coredns/releases. After you find the latest patch version just run following command. For e.g. `export latest_coredns="1.6.5"`.
+
+**Note**: There is no `v` prefixed to the version.
+
+```
+export latest_coredns="<latest patch version of coredns>"
+```
+
+* For `calico` check the latest patch version.
+
+Find your current version:
+
+```
+kubectl -n kube-system get ds calico-node -o jsonpath='{.spec.template.spec.containers[0].image}' | cut -d':' -f2-
+```
+
+You can see the latest patch version on the Calico release page: https://github.com/projectcalico/calico/releases. After you find the latest patch version just run following command. For e.g. `export latest_calico="v3.8.4"`.
+
+```
+export latest_calico="<latest patch version of calico>"
+```
+
+## Upgrade Control Plane
+
+While making following changes you can have a terminal where you see all the events from `kube-system` namespace:
+
+```
+kubectl -n kube-system get events -w
+```
+
+### kube-apiserver
+
+```
+kubectl set image daemonset -n kube-system kube-apiserver kube-apiserver="k8s.gcr.io/hyperkube:${latest_kube}"
+```
+
+You can verify that apiserver is back by running following command:
+
+```
+kubectl get daemonset -n kube-system kube-apiserver
+```
+
+### kube-scheduler
+
+```
+kubectl set image deployment -n kube-system kube-scheduler kube-scheduler="k8s.gcr.io/hyperkube:${latest_kube}"
+```
+
+You can verify that scheduler is back by running following command:
+
+```
+kubectl get deployment -n kube-system kube-scheduler
+```
+
+### kube-controller-manager
+
+```
+kubectl set image deployment -n kube-system kube-controller-manager kube-controller-manager="k8s.gcr.io/hyperkube:${latest_kube}"
+```
+
+You can verify that controller manager is back by running following command:
+
+```
+kubectl get deployment -n kube-system kube-controller-manager
+```
+
+### kube-proxy
+
+```
+kubectl set image daemonset -n kube-system kube-proxy kube-proxy="k8s.gcr.io/hyperkube:${latest_kube}"
+```
+
+You can verify that kube-proxy is back by running following command:
+
+```
+kubectl get daemonset -n kube-system kube-proxy
+```
+
+### CoreDNS
+
+```
+kubectl set image deployment -n kube-system coredns coredns="k8s.gcr.io/coredns:${latest_coredns}"
+```
+
+```
+kubectl get deployment -n kube-system coredns
+```
+
+### Calico
+
+```
+kubectl set image daemonset -n kube-system calico-node calico-node="quay.io/calico/node:${latest_calico}" install-cni="quay.io/calico/cni:${latest_calico}"
+```
```

Comment:
Does calico have any docs saying we can upgrade like this? Or is just obvious that this will work? :)

Reply:
I am not sure of the calico docs, I assume that this will just work.

Reply:
Ok, maybe we can double check with someone else. Andy/Iago probably know right away?

Reply:
@iaguis your help would be helpful here.

Is it safe to assume that if we update the calico from one **patch version** to another without making any changes to the config, it will work just fine?

Reply:
> your help would be helpful

happy to help with my helpful help :D

From what I understand Calico is backwards compatible (@ahrkrak can correct me if I'm wrong), even more so with patch releases so I see no problem there.

Reply:
Perfect, thanks!